src/util/ha-url-sync.html

```diff
@@ -31,7 +31,11 @@
       if (newHass.moreInfoEntityId) {
         if (DEBUG) console.log('pushing state');
         history.pushState(null, null, window.location.pathname);
-      } else {
+        // We keep track of where we opened moreInfo from so that we don't
+        // pop the state when we close the modal if the modal has navigated
+        // us away.
+        this.moreInfoOpenedFromPath = window.location.pathname;
```

Comment:
Move this to before pushState? Could there be some callback on it that changes location?

Reply:
Updated it. Shouldn't happen right now but order doesn't matter, so why not.